File: Programmers/Swift/타겟 넘버.swift
import Foundation

var res = 0

func dfs(_ n:[Int], _ t:Int, _ idx:Int, _ sum: Int) {
    if idx == n.count {
        if sum == t {
            res += 1
        }
        return
    }
    dfs(n, t, idx+1, sum+n[idx])
    dfs(n, t, idx+1, sum-n[idx])
}

func solution(_ numbers:[Int], _ target:Int) -> Int {
    dfs(numbers, target, 0, 0)
    return res
}

File: Programmers/C++/타겟 넘버.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#define FOR(i,n) for(int i=0;i<(n);++i)
#define FORS(i,s,n) for(int i=(s);i<(n);++i)

using namespace std;

int res;

void dfs(vector<int>& n, int target, int sum, int idx) {
    if(idx==n.size()) {
        if(sum==target) ++res;
        return;
    }
    dfs(n,target,sum+n[idx],idx+1);
    dfs(n,target,sum-n[idx],idx+1);
}

int solution(vector<int> n, int target) {
    dfs(n,target,0,0);
    
    return res;
}
